core/src/main/scala/filodb.core/Messages.scala

```diff
@@ -29,3 +29,5 @@ final case class NotFoundError(what: String) extends Exception(what)
 final case class StorageEngineException(t: Throwable) extends Exception(t)
 final case class MetadataException(t: Throwable) extends Exception(t)
 final case class SystemLimitsReachedException(msg: String) extends Exception(msg)
+final case class QueryTimeoutException(queryTime: Long, className: String) extends
```

Comment:
rename `className` param as `timedOutAt` and some docs such as `placeholder for additional context on where the query was when it timed out`